trio/_core/_io_kqueue.py

```diff
@@ -10,7 +11,7 @@
 from ._run import _public
 from ._wakeup_socketpair import WakeupSocketpair
 
-assert sys.platform != "linux" and sys.platform != "win32"
+assert not TYPE_CHECKING or sys.platform != "linux" and sys.platform != "win32"
```

Comment:
Please put parens around the `and` clause here. I know `or` is lower precedence, but I had to look it up :-)

And feel free to merge once you've done that

Reply:
Fair enough :) Done, and merged